api/utils/keys/policy.go

```diff
@@ -72,10 +72,14 @@ func NewPrivateKeyPolicyError(p PrivateKeyPolicy) error {
 // ParsePrivateKeyPolicyError checks if the given error is a private key policy
 // error and returns the contained unmet PrivateKeyPolicy.
 func ParsePrivateKeyPolicyError(err error) (PrivateKeyPolicy, error) {
+	if err == nil {
```

Comment:
This seems odd. Why would we ever call this with a nil error? If we do, aren't we effectively converting the "no error" case into an error?

Reply:
I suspect this is getting around the potential nil pointer error on line 80:
```
	subMatches := privateKeyPolicyErrRegex.FindStringSubmatch(err.Error())
```

So they effectively represent the same error case.

Reply:
I can remove the check if we'd like, a nil pointer exception would only occur on developer error, as a nil error should never be passed to the function.